src/puppetlabs/pcp/client.clj

```diff
@@ -257,27 +278,34 @@
   (let [{:keys [cert private-key cacert]} ssl-files]
     (ssl-utils/pems->ssl-context cert private-key cacert)))
 
-;; private helpers for the ssl/websockets setup
-(s/defn ^:private make-ssl-context :- SslContextFactory
-  "Returns an SslContextFactory that does client authentication based on the
-  client certificate named"
+;; public helpers for the ssl/websockets setup, also used by pcp-broker's test client
+(s/defn ^:public make-ssl-context :- SslContextFactory$Client
+  "Returns an SslContextFactory$Client that does client authentication based on the
+  client certificate named."
   [params]
-  (let [factory (SslContextFactory.)
+  (let [factory (SslContextFactory$Client.)
         ssl-context (get-ssl-context (:ssl-context params))]
     (.setSslContext factory ssl-context)
-    (.setNeedClientAuth factory true)
+    ;;(.setNeedClientAuth factory true) ;; not available with $Client, problem?
     (.setEndpointIdentificationAlgorithm factory "HTTPS")
     factory))
 
-(s/defn ^:private make-websocket-client :- WebSocketClient
-  "Returns a WebSocketClient with the correct SSL context"
-  [ssl-context :- SslContextFactory max-message-size :- (s/maybe s/Int)]
-  (let [client (WebSocketClient. ssl-context)]
-    (if max-message-size
-      (.setMaxTextMessageSize (.getPolicy client) max-message-size))
+(s/defn ^:public make-websocket-client :- WebSocketClient
```

Comment:
I moved the above two helper functions to public to faciliate pcp-broker tests (will see why when I get that PR up), perhaps not desirable ultimately though it is a pretty harmless change for now.